### DISCUSSION TITLE
Dirty hack for the cover page of a book.

### DISCUSSION THREAD

**Discussion by no-vici** [Omikhleia/resilient.sile#23] [Ideas]
This is my current workflow for the cover page:

1. The cover page is designed externally.
2. Create a cover page holder in your book.
3. Extract the dump_data of the pdf by [pdftk](https://www.pdflabs.com/tools/pdftk-the-pdf-toolkit/)
4. Replace the cover page with the new one.
5. Import the dump_data back into the new pdf from the last step.

I have a messy [ebook-builder](https://drive.google.com/drive/folders/1JcrEJCZjTOEJWqBCYE6mkSmu55TUkZkg?usp=share_link) that you can have a look, and hope that it will  save you some time building one within the perspective of SILE while waiting for the solution from the author of resilient.sile.

@Omikhleia, would you mind taking a look at my script to see if the style paradigm could have a "cover page" option. I know that you are still working on the front matter in the background.

**Comment by Omikhleia** [comment]
For the record, the next "feature" release of resilient (Oct. or Nov. 2023) will have provisions (with master documents) to "automatically" include cover images and also generate half-title page, title-page and end-papers based on document metadata. I'll link to the PR when I start pushing things forward.

**Reply by Omikhleia** [reply]
Link to the PR: https://github.com/Omikhleia/resilient.sile/pull/57

**Reply by Omikhleia** [reply]
"Book matters" went in release [2.2.0](https://github.com/Omikhleia/resilient.sile/releases/tag/v2.2.0)

1. The cover pages are designed externally (front and back cover images).
2. The "master document" has provisions for including these cover pages
3. Back cover support has provision for including back cover text and the ISBN, etc.

**Reply by no-vici** [reply]
I must try this new feature. 

Thank you for all the hard work!

**Comment by Omikhleia** [comment]
Marked as closed, since we have a standard solution.
